src/main/marketplace/tax-treaties/tax-treaties-service.js

```diff
@@ -13,6 +13,17 @@ export const TAX_TREATIES_API_ENDPOINT = `${config.airtableBaseUrl}TaxTreaties${
 	isDevMode() ? 'Dev' : ''
 }`;
 
+// Helper function to evenly distribute in time expensive background processing.
+// This avoids blocking the event loop, and should be used when we are expecting
+// to parse and process a very large number of (non time-critical) items
+const executeAsync = (idx, fx) =>
+	new Promise((resolve, reject) =>
+		setTimeout(() => {
+			const item = fx();
+			resolve(item);
+		}, idx * 10)
```

Comment:
index * 10 ?

you don't need to worry about finish time since you use `Promise.all` anyway, also all timers are executed via queue so your order will be preserved anyway if you set it to the same timeout. Moreover, you don't need to worry about timeout if you use setImmediate